File: src/utils/dependencies.ts
// Types
import type { PackageManager } from '@/types/package-mangers'

// Constants
import { INSTALLATION_COMMANDS } from '@/constants/dependencies'

// Third-Party libraries
import promiseSpawn from '@npmcli/promise-spawn'

// NodeJS
import process from 'node:process'

// Utils
import { writeMessage } from './console'
import { getErrorMessage } from './errors'

interface Props {
  packageManagerToUse: PackageManager
  packagesToInstall: string | string[]
}

export async function installDependencies({
  packageManagerToUse,
  packagesToInstall
}: Props) {
  try {
    const installationCommand = INSTALLATION_COMMANDS[packageManagerToUse]

    writeMessage({
      type: 'info',
      message: `Installing dependencies using: ${packageManagerToUse}...`
    })

    await promiseSpawn(
      packageManagerToUse,
      [installationCommand, packagesToInstall, '-D'].flat()
    )

    writeMessage({
      type: 'success',
      message: 'Dependencies installed successfully'
    })
  } catch (error) {
    writeMessage({
      type: 'error',
      message: getErrorMessage('Dependencies')
    })
    process.exit(1)
  }
}

File: src/utils/dependencies.go
package utils

import (
	"os"
	"os/exec"
	"stylelintbc/src/constants"
	"stylelintbc/src/types"
)

type InstallDependenciesProps struct {
	PackageManagerToUse types.PackageManager
	PackagesToInstall   []string
}

func InstallDependencies(props InstallDependenciesProps) {
	notificationMessage := "Installing dependencies using: " + props.PackageManagerToUse + "..."

	WriteMessage(WriteMessageProps{
		Type:    "info",
		Message: string(notificationMessage),
	})

	installationCommand := constants.INSTALLATION_COMMANDS[props.PackageManagerToUse]

	commandArgs := installationCommand

	for _, pkg := range props.PackagesToInstall {
		commandArgs += " " + pkg
	}

	commandArgs += " " + "-D"

	println("Command arguments: " + commandArgs)

	cmd := exec.Command(
		string(props.PackageManagerToUse),
		commandArgs,
	)

	_, err := cmd.CombinedOutput()

	if err != nil {
		WriteMessage(WriteMessageProps{
			Type:    "error",
			Message: string(GetErrorMessage("Dependencies")),
		})

		os.Exit(1)
	}

	WriteMessage(WriteMessageProps{
		Type:    "success",
		Message: "Dependencies installed successfully",
	})
}
